tests/target/issue-4119.rs

```diff
@@ -0,0 +1,177 @@
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 if 1 = 0 /* check if present */ => ReturnCode::SUCCESS,
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 if /* check if present */ 1 = 0 => ReturnCode::SUCCESS,
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+// With long comments
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */ => {
+            ReturnCode::SUCCESS
+        }
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */ if 1 = 0 => {
+            ReturnCode::SUCCESS
+        }
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 if 1 = 0 /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */
+        => ReturnCode::SUCCESS,
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        0 if /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */ 1 = 0 => {
+            ReturnCode::SUCCESS
+        }
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+// Multiline Pattern
+
+fn foo() -> ReturnCode {
+    match command_num {
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        | BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
+        /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */ => {
+            ReturnCode::SUCCESS
+        }
+        // Check is sensor is correctly connected
+        1 => {
+            let x = 0;
+            ReturnCode::EBUSY
+        }
+        _ => ReturnCode::ENOSUPPORT,
+    }
+}
+
+fn foo() -> ReturnCode {
+    match command_num {
+        AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+        | BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
+        /* Loooooooooooooooooooooooooooooooooooooooooonnnnnnnnnnng comment */
+        if 1 = 0 =>
```

Comment:
I think we'd need to do something different with the indentation on these, but to be honest I'm not sure what would make sense as these go into some rather extreme scenarios. I think we'd probably still want the guard to be block indented, but that still leaves the question of what to do with the comment